react/MinicartContext.tsx

```diff
@@ -57,7 +70,9 @@ const MinicartContextProvider: FC<Props> = ({
     variation: resolvedVariation,
     open: false,
     hasBeenOpened: false,
-    openOnHover: resolvedVariation !== 'popup' ? false : openOnHover,
+    openOnHoverProp: openOnHover,
+    openOnHoverBehavior:
+      resolvedVariation === 'popup' && !isMobile ? openOnHover : false,
```

Comment:
Those variables names are confusing..

Can we think of better names? I would suggest some, but I don't understand the difference.

Reply:
Agreed, I could not think of much better one right off the bat.
The `openOnHoverProp` is the original value provided by the user via the `openOnHover` prop from the API. While the `openOnHoverBehavior` is an internal state variable that acts as a toggle between the *normal* behavior (open and close on click) and the *hover* behavior (open on hover and close on mouse leave event)

Reply:
That is needed in order to achieve this:
> If the user CLICKS on the minicart icon, then it should behave as usual and remains open after the user removes the mouse of it.